### PR TITLE
Simulation line

### PR DESCRIPTION
**Pull request by maggieguo711**
Change previous 5 paths to green gradual gradient

### CODE REVIEW

**Review by jackiepeng15 — CHANGES_REQUESTED** [review]
Hey Maggie! I'd appreciate it a lot if you could add more comments about what each block of code is doing, because right now when I'm looking at it it's hard for me to figure out what's going on and if I were to have to work on this after you it would take me a lot of effort to figure out what everything means. Thank you!

File: StaticGUI.py [hunk 1]
```diff
@@ -154,15 +154,61 @@ def runSimulation(self, smoothPath):
...
+        self.set_of_prev_path = []
 
     def drawPath(self):
+        # change previous 5 path to rainbow colors
+
+        color = 'gray1'
+        if self.prev_line_id:
+            self.set_of_prev_path.append(self.prev_line_id)
+
+        if self.set_of_prev_path:
```
Comment (ShivalikaGupta) [inline StaticGUI.py:174]: Maybe you can put the colors in a list called colors and then shorten these lines of code to a for loop. Replace the indexes inside self.set_of_prev_path with the variable in the for loop (i.e the variable i for example) and call colors[i] inside the if statement.

File: StaticGUI.py [hunk 2]
```diff
@@ -154,15 +154,61 @@ def runSimulation(self, smoothPath):
...
+                color = 'none'
+
+        if color == 'none':
+            for fst_id in self.set_of_prev_path[0]:
+                self.canvas.delete(fst_id)
+            self.set_of_prev_path.pop(0)
+            color = 'gray1'
+
+        if len(self.set_of_prev_path) > 5:
+            for id in self.set_of_prev_path[0]:
```
Comment (ShivalikaGupta) [inline StaticGUI.py]: This can also be a for loop using the colors list.

File: StaticGUI.py [hunk 3]
```diff
@@ -154,15 +154,22 @@ def runSimulation(self, smoothPath):
...
+        self.prev_line_id = []
+
         idx = 1
         while idx < len(self.path):
             x1 = self.path[idx - 1].x / tile_scale_fac
             y1 = self.path[idx - 1].y / tile_scale_fac
             x2 = self.path[idx].x / tile_scale_fac
             y2 = self.path[idx].y / tile_scale_fac
-            self.canvas.create_line(x1, y1, x2, y2, fill="#339933")
+            canvas_id = self.canvas.create_line(x1, y1, x2, y2, fill="#8cf075")
```
Comment (Z01iC) [inline StaticGUI.py]: Not a bad idea to define color # in CONSTS.PY, so for example GREENPATH = "#8cf075"

File: StaticGUI.py [hunk 4]
```diff
@@ -154,15 +154,22 @@ def runSimulation(self, smoothPath):
 class SmoothedPathGUI(MapPathGUI):
     def __init__(self, master, inputMap, path):
         super().__init__(master, inputMap, path)
+        self.prev_line_id = []
```
Comment (Z01iC) [inline StaticGUI.py:157]: id rename to prev_line_ids

whenever you have a collection of something it is good practice to make the varaible name plural

File: StaticGUI.py (hunk 1 above)
Comment (Z01iC) [reply StaticGUI.py:174]: How about this....
color = None
colors = ['#347800', '#347800'...]
if(self.prev_line_id in self.set_of_orev_path):
    color = colors[self.set_of_orev_path.index(self.prev_line_id)

Comment (Z01iC) [reply StaticGUI.py:174]: I'd also make the colors lists something you put in consts.py if you use this. All the wonky hashtags should be given word names like GREEN0, GREEN1... or something intuitive

**Pull request merged**